packages/idempotency/src/persistence/DynamoDBPersistenceLayer.ts

```diff
@@ -230,14 +223,6 @@ class DynamoDBPersistenceLayer extends BasePersistenceLayer {
     );
   }
 
-  private getClient(): DynamoDBClient {
-    if (!this.client) {
-      this.client = new DynamoDBClient(this.clientConfig);
```

Comment:
The idea here was to initialize the client lazily.

Couldn't we have added the middleware at this point?

```ts
if (!this.client) {
  this.client = new DynamoDBClient(this.clientConfig);
  this.client.middleware.add(...);
}
```

and do the same in the `constructor` with the one that they pass us?

Reply:
I wanted to have one place in constructor to do that. Is the performance noticeable?

Reply:
I think it's not, let's go with your version.

Let's streamline the constructor altogether. Basically let's do the same as the Parameters util & remove the `private client!` assertion, since we know it'll always be set before the constructor returns.

Reply:
Alright, I'll revert to the previous version with lazy init.